### PR TITLE
Add Prettier, add support for `@tool`, `@icon`, `@export` & co, `@onready`, typed signals, `%node` 

### PR DESCRIPTION
**Pull request by Cretezy**
Closes https://github.com/PrestonKnopp/tree-sitter-gdscript/issues/9

Hi! I tried to originally write code without a formatter, however it was difficult to match your style. I've added Prettier to the project due to that.

I've seperated this PR into commits, with the [Run Prettier](https://github.com/PrestonKnopp/tree-sitter-gdscript/commit/d3c19f40bd97460c3a47f182932e9082acff09e5) one being just `npm run format`. The [last commit](https://github.com/PrestonKnopp/tree-sitter-gdscript/commit/08056ec2fa336d5419d3476b525700b8fcf41978) is the one responsible for the main code changes. 

Thanks!

### CODE REVIEW

**Comment by Cretezy** [comment]
Examples of it working:

image
image
image

**Comment by PrestonKnopp** [comment]
Awesome, thanks for the PR.

Formatting, typed signal parameters, and get unique node looks good.

What are your thoughts on parsing `@annotation` as its own ast? For example:

```
(annotation (identifier) (argument_list))
```

Also, looking through this I see that the current `export_variable_statement` ast is not accurate. Var stmts can be exported and onready. I'll add a new issue for this.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-package-lock.json
```
Comment (PrestonKnopp) [inline .gitignore:1]: I ignored package-lock for a reason that I forgot now. If only I had a better commit message than "updated gitignore".

Comment (Cretezy) [reply .gitignore:1]: I've had no issue with `package-lock.json`, maybe give it another try? I'm OK with removing this, but ideally we should have it

**Comment by PrestonKnopp** [comment]
Yes, remove the annotations and then I'll merge this right away. Thanks!

**Comment by Cretezy** [comment]
@PrestonKnopp Done! For reference to be readded:

- `@tool`
- `@icon`
- `@onready`
- `@export` and `@export_*`
- `@rpc` (wasn't added here)

**Comment by PrestonKnopp** [comment]
Awesome, thank you!

**Pull request merged**